internal/integration-corpus-generator/assets.go

```diff
@@ -0,0 +1,119 @@
+
+	"github.com/elastic/elastic-package/internal/builder"
+)
+
+// GetGoTextTemplate returns the gotext template of a package's data stream
+func (c *Client) GetGoTextTemplate(packageName, dataStreamName string) ([]byte, error) {
+	assetsSubFolder := fmt.Sprintf("%s.%s", packageName, dataStreamName)
+	gotextTemplateAssetName := fmt.Sprintf("%s.gotext.log", dataStreamName)
+	statusCode, respBody, err := c.get(fmt.Sprintf("%s/%s", assetsSubFolder, gotextTemplateAssetName))
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get gotext template")
+	}
+
+	if statusCode != http.StatusOK {
+		return nil, fmt.Errorf("could not get gotext template; API status code = %d; response body = %s", statusCode, respBody)
+	}
+
+	return respBody, nil
+}
+
+// GetConf returns the genlib.Config of a package's data stream
+func (c *Client) GetConf(packageName, dataStreamName string) (genlib.Config, error) {
+	assetsSubFolder := fmt.Sprintf("%s.%s", packageName, dataStreamName)
+	confYamlAssetName := fmt.Sprintf("%s.conf.yml", dataStreamName)
+
+	statusCode, respBody, err := c.get(fmt.Sprintf("%s/%s", assetsSubFolder, confYamlAssetName))
+	if err != nil {
+
+		return genlib.Config{}, errors.Wrap(err, "could not get config yaml")
+	}
+
+	if statusCode != http.StatusOK {
+		return genlib.Config{}, fmt.Errorf("could not get config yaml; API status code = %d; response body = %s", statusCode, respBody)
+	}
+
+	cfg, err := config.LoadConfigFromYaml(respBody)
+	if err != nil {
+		return genlib.Config{}, errors.Wrap(err, "could not load  config yaml")
+	}
+
+	return cfg, nil
+}
+
+// GetFields returns the genlib.Config of a package's data stream
+func (c *Client) GetFields(packageName, dataStreamName string) (genlib.Fields, error) {
+	assetsSubFolder := fmt.Sprintf("%s.%s", packageName, dataStreamName)
+	fieldsYamlAssetName := fmt.Sprintf("%s.fields.yml", dataStreamName)
+
+	statusCode, respBody, err := c.get(fmt.Sprintf("%s/%s", assetsSubFolder, fieldsYamlAssetName))
+	if err != nil {
+		return genlib.Fields{}, errors.Wrap(err, "could not get config yaml")
+	}
+
+	if statusCode != http.StatusOK {
+		return genlib.Fields{}, fmt.Errorf("could not get config yaml; API status code = %d; response body = %s", statusCode, respBody)
+	}
+
+	fieldsDefinitionPath, err := writeFieldsYamlFile(respBody, packageName, dataStreamName)
+	if err != nil {
+		return genlib.Fields{}, errors.Wrap(err, "could not load fields yaml")
+	}
+
+	ctx := context.Background()
+	flds, err := fields.LoadFieldsWithTemplate(ctx, fieldsDefinitionPath)
+	if err != nil {
+		return genlib.Fields{}, errors.Wrap(err, "could not load fields yaml")
+	}
+
+	return flds, nil
+}
+
+func tmpGenlibDir() (string, error) {
+	buildDir, err := builder.BuildDirectory()
+	if err != nil {
+		return "", errors.Wrap(err, "locating build directory failed")
+	}
+	return filepath.Join(buildDir, "genlib"), nil
+}
+
+func writeFieldsYamlFile(fieldsYamlContent []byte, packageName, dataStreamName string) (string, error) {
```

Comment:
this is currently needed because we don't have anything else than `fields.LoadFieldsWithTemplate` (requiring a path to a local file) to loads `Fields` from custom content. I will refactor/add a method that accepts `[]byte` instead in the dependency package, so that we can skip this